src/jit/codegenxarch.cpp

```diff
@@ -6789,30 +6805,60 @@ void         CodeGen::genJumpKindsForTreeLongLo(GenTreePtr    cmpTree,
 //                      cmp loOp1,loOp2
 //                      label: setne
 //
-//    GT_LT             cmp hiOp1,hiOp2         If hiOp1 is greater than hiOp2, the entire compare
-//                      ja label                is false. If hiOp1 is less than hiOp2, the entire
-//                      jne label               compare is true.
-//                      cmp loOp1,loOp2
+//    GT_LT; unsigned   cmp hiOp1,hiOp2         If hiOp1 is not equal to hiOp2, the entire compare
+//                      jne label               is false. Otherwise, we need to compare the lo
```

Comment:
The comment here (and below), says "If hiOp1 is not equal to hiOp2, the entire compare is false". However, it's not that it's false, its that the result is determined, and the flags are properly set for the future 'set' instruction.